dfusion_rust_core/src/models/util.rs

```diff
@@ -220,3 +221,7 @@ pub fn read_amount(bytes: &[u8; 12]) -> u128 {
 
     BigEndian::read_u128(bytes.as_slice())
 }
+
+pub fn map_from_slice(arr: &[(u16, u128)]) -> HashMap<u16, u128> {
```

Comment:
nit: can this be generics?

Reply:
Ya sure! Great thinking! What about the suggestion to make a macro?

Reply:
Done!